### ISSUE TITLE
`inline if` with `null` comparison fails to expand

### ISSUE THREAD

**Issue by japgolly** [lampepfl/dotty#12180]
## Compiler version

3.0.0-RC3

## Minimized code

```Scala
inline def n = null
inline def f = inline if n == null then 0 else 1
inline def m = inline n match { case null => 0; case _ => 1 }

@main def main =
  println(f) // error
  println(m)
```

## Output

```
[error] 6 |  println(f)
[error]   |          ^
[error]   |Cannot reduce `inline if` because its condition is not a constant value: (null:Null).==(null)
[error]   | This location contains code that was inlined from x.scala:2
[error] one error found
```

## Expectation

Success.

**Comment by japgolly** [comment]
Oh, it's not just `null` on `null`, it seems it's anything and `null`:

```scala
inline def n = "im a string"
inline def f = inline if n == null then 0 else 1
inline def g = inline if null == n then 0 else 1
inline def m = inline n match { case null => 0; case _ => 1 }

@main def main =
  println(f)
  println(g)
  println(m)
```

```
[error] 7 |  println(f)
[error]   |          ^
[error]   |Cannot reduce `inline if` because its condition is not a constant value: ("im a string":String).==(null)

[error] 8 |  println(g)
[error]   |          ^
[error]   |Cannot reduce `inline if` because its condition is not a constant value: null.==("im a string":String)
```

**Comment by nicolasstucki** [comment]
This was fixed in #12080. It will be in `3.0.1`.

**Comment by japgolly** [comment]
Ah of course, whoops. Thanks @nicolasstucki
